### PR TITLE
refactoring the tests and removing cargo dependencies

### PR DESCRIPTION
**Pull request by sendilkumarn**
This addresses #13 

removing the entire cargo things from the application and move the samples that uses rust into examples folder and then relaying on the rustc rather than cargo

### CODE REVIEW

File: bors.toml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-status = [
-  "continuous-integration/travis-ci/push",
-]
```
Comment (killercup) [inline bors.toml]: Why'd you kill bors? 😮

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Some experiments after seeing Rust PR #45905.",
   "main": "do-the-wasm.js",
   "scripts": {
-    "prestart": "make",
+    "rust-sample": "make --directory=examples/rust",
```
Comment (killercup) [inline package.json]: Can you run this on CI and make sure the sample tests are executed?

File: tests/hello-example.js [hunk 3]
```diff
@@ -1,71 +1,188 @@
 const { test } = require("ava");
-const { instantiate } = require("../src/wasm");
+const loadRust = require("../src/compile");
 const wrap = require("../src/wrap");
 
-test.beforeEach(async (t) => {
-  t.context.i = await instantiate("./hello-wasm.wasm");
-});
+const allocFunction = `
```
Comment (killercup) [inline tests/hello-example.js:5]: 👍

Comment (killercup) [reply tests/hello-example.js:5]: Feel free to add use `std::mem; use std::os::raw::{c_char, c_void};`  as well

File: bors.toml (hunk 1 above)
Comment (sendilkumarn) [reply bors.toml]: Well that is a mistake, sorry :) `.toml` 😬

File: tests/hello-example.js (hunk 3 above)
Comment (sendilkumarn) [reply tests/hello-example.js:5]: yeah could have done that 👍

File: package.json (hunk 2 above)
Comment (sendilkumarn) [reply package.json]: this is not a test any more, it is an example. we can run this step on the CI... Do you want to have this as a test ?

Comment (killercup) [reply package.json]: I want to have every single line of code in this repo tested on CI :) Which includes `examples/rust/test.js`

Comment (sendilkumarn) [reply package.json]: can't agree more... 👍

File: tests/hello-example.js (hunk 3 above)
Comment (badboy) [reply tests/hello-example.js:5]: `std::meme`

**Review by killercup — APPROVED** [review]
LGTM!

bors r+

**Pull request merged**